src/hostname/hostnamed.c

```diff
@@ -1126,9 +1150,47 @@ static int method_get_hardware_serial(sd_bus_message *m, void *userdata, sd_bus_
         return sd_bus_send(NULL, reply, NULL);
 }
 
+static int method_get_bios_version(sd_bus_message *m, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_free_ char *biosversion = NULL;
+        Context *c = userdata;
+        int r;
+
+        assert(m);
+        assert(c);
+
+        r = bus_verify_polkit_async(
+                        m,
+                        CAP_SYS_ADMIN,
+                        "org.freedesktop.hostname1.get-bios-version",
+                        NULL,
+                        false,
+                        UID_INVALID,
+                        &c->polkit_registry,
+                        error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
+
+        r = get_bios_version(&biosversion);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_new_method_return(m, &reply);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_append(reply, "s", biosversion);
+        if (r < 0)
+                return r;
+
+        return sd_bus_send(NULL, reply, NULL);
+}
```

Comment:
and here, too, would prefer if we could share some code here with the hardware serial getter function. i.e. maybe use a function pointer parameter to choose whether to call the existing get_hardware_serial() or get_bios_version().

See how property_get_hardware_property() does this with property_get_hardware_vendor() and property_get_hardware_model(), for inspiration

Reply:
I think we can remove polkit authentication as bios version can be read without root permission . We can get bios version without root permission using "/sys/class/dmi/id/bios_version" 
@poettering it will be helpful, if you can relay your comment on this .. Thank you